plugins/modules/clustering/nomad/nomad_job_info.py

```diff
@@ -0,0 +1,152 @@
+    - List Nomad jobs.
+requirements:
+  - python-nomad
+options:
+    host:
+      description:
+        - FQDN of Nomad server.
+      required: true
+      type: str
+    use_ssl:
+      description:
+        - Use TLS/SSL connection.
+      type: bool
+      default: true
+    timeout:
+      description:
+        - Timeout (in seconds) for the request to Nomad.
+      type: int
+      default: 5
+    validate_certs:
+      description:
+        - Enable TLS/SSL certificate validation.
+      type: bool
+      default: true
+    client_cert:
+      description:
+        - Path of certificate for TLS/SSL.
+      type: path
+    client_key:
+      description:
+        - Path of certificate's private key for TLS/SSL.
+      type: path
+    namespace:
+      description:
+        - Namespace for Nomad.
+      type: str
+    token:
+      description:
+        - ACL token for authentification.
+      type: str
+    name:
+      description:
+        - Name of job for Get info.
+      type: str
+notes:
+  - C(check_mode) is supported
+seealso:
+  - name: Nomad jobs documentation
+    description: Complete documentation for Nomad API jobs.
+    link: https://www.nomadproject.io/api-docs/jobs/
+'''
+
+EXAMPLES = '''
+- name: Get info for job awx
+  community.general.nomad_job:
+    host: localhost
+    name: awx
+
+- name: List Nomad jobs
+  community.general.nomad_job:
+    host: localhost
+
+'''
+
+
+import os
+import json
+
+from ansible.module_utils.basic import AnsibleModule, missing_required_lib
+from ansible.module_utils._text import to_native
+
+import_nomad = None
+try:
+    import nomad
+    import_nomad = True
+except ImportError:
+    import_nomad = False
+
+
+def run():
+    module = AnsibleModule(
+        argument_spec=dict(
+            host=dict(required=True, type='str'),
+            use_ssl=dict(type='bool', default=True),
+            timeout=dict(type='int', default=5),
+            validate_certs=dict(type='bool', default=True),
+            client_cert=dict(type='path', default=None),
+            client_key=dict(type='path', default=None),
+            namespace=dict(type='str', default=None),
+            token=dict(type='str', default=None, no_log=True),
+            name=dict(type='str', default=None)
+        ),
+        supports_check_mode=True
+    )
+
+    if not import_nomad:
+        module.fail_json(msg=missing_required_lib("python-nomad"))
+
+    certificate_ssl = (module.params.get('client_cert'), module.params.get('client_key'))
+
+    nomad_client = nomad.Nomad(
+        host=module.params.get('host'),
+        secure=module.params.get('use_ssl'),
+        timeout=module.params.get('timeout'),
+        verify=module.params.get('validate_certs'),
+        cert=certificate_ssl,
+        namespace=module.params.get('namespace'),
+        token=module.params.get('token')
+    )
+
+    if module.params.get('name'):
+
+        try:
+            result = nomad_client.job.get_job(module.params.get('name'))
```

Comment:
I'm not sure what the API returns in both cases, but the return type should not depend on whether `name` was specified or not. So this probably needs to be put into a list, or something like that.

Also you need to document what happens if the job does not exist. Returning an empty list would probably be the expected behavior.

Reply:
@felixfontein ok i inderstand, so listing job can be write in nomad_jobs_list_info, and nomad_job_info get selected info of one job ?

Reply:
Many _info modules work like this:
1. If `name` is not specified, return a list of all objects.
2. If `name` is specified, filter the list by the objects called `name` (i.e. usually return either 1 or 0 objects).

Alternatively, you can have two modules, `_info` (returns info on one specific object) and `_list_info` (lists all objects).